test/integration/garbagecollector/garbage_collector_test.go

```diff
@@ -265,12 +266,6 @@ func setupWithServer(t *testing.T, result *kubeapiservertesting.TestServer, work
 	}
 	syncPeriod := 5 * time.Second
 	startGC := func(workers int) {
-		go wait.Until(func() {
```

Comment:
This was the cause of the timing issues I was stuck on forever. I’m a bit stumped on why we are continuously resetting the REST mapper every syncPeriod. Removing it didn’t break anything (and fixed the CRD test issues) and the PR it came from wasn’t very revealing, so I figured it was a vestige of something no longer relevant. Let me know if you think I’m missing something here.

Reply:
effectively, this is forcing us to grab new data from the apiserver.  The idea here is that refetching the discover doc allows us to see that the CRD has been removed (since the corresponding GVK is gone from discovery), so we can see that we need to process its removal.